packages/ui/src/models/camel/kamelet-resource.ts

```diff
@@ -1,56 +1,39 @@
-import { KameletVisualEntity } from './../visualization/flows/kamelet-visual-entity';
-import { getCamelRandomId } from '../../camel-utils/camel-random-id';
+import set from 'lodash.set';
+import { TileFilter } from '../../public-api';
 import { IKameletDefinition } from '../kamelets-catalog';
 import { AddStepMode } from '../visualization/base-visual-entity';
+import { CamelRouteVisualEntity } from '../visualization/flows';
 import { FlowTemplateService } from '../visualization/flows/flow-templates-service';
+import { CamelComponentFilterService } from '../visualization/flows/support/camel-component-filter.service';
+import { CamelRouteVisualEntityData } from '../visualization/flows/support/camel-component-types';
+import { KameletVisualEntity } from './../visualization/flows/kamelet-visual-entity';
 import { CamelKResource } from './camel-k-resource';
 import { SourceSchemaType } from './source-schema-type';
-import { CamelRouteVisualEntityData } from '../visualization/flows/support/camel-component-types';
-import { TileFilter } from '../../public-api';
-import { CamelComponentFilterService } from '../visualization/flows/support/camel-component-filter.service';
 
 export class KameletResource extends CamelKResource {
-  private kamelet;
   private flow: KameletVisualEntity;
 
   constructor(kamelet?: IKameletDefinition) {
     super(kamelet);
-    const kameletId = (kamelet?.metadata?.name as string) ?? getCamelRandomId('kamelet');
 
-    if (kamelet) {
-      this.kamelet = kamelet;
-    } else {
-      this.kamelet = this.resource as IKameletDefinition;
-      this.kamelet.kind = SourceSchemaType.Kamelet;
-      this.kamelet.spec = {
-        definition: {
-          title: kameletId,
-          type: 'source',
-        },
-        dependencies: [],
-        template: {
-          from: {
-            id: getCamelRandomId('from'),
-            uri: 'kamelet:source',
-            steps: [],
-          },
-          beans: {},
-        },
-      };
+    if (!kamelet) {
+      this.resource = FlowTemplateService.getFlowTemplate(this.getType());
     }
 
-    this.flow = new KameletVisualEntity(this.kamelet);
+    this.flow = new CamelRouteVisualEntity({
```

Comment:
Wouldn't `this.flow = new KameletVisualEntity(this.resource as IKameletDefinition);` in the end do the same? (might be missing something here, so sorry if this is stupid question)

Reply:
besides this, the PR looks and works good

Reply:
Ah, I think you're right, it might be a leftover from a test, I'll update it.

Reply:
@tplevko, I've replaced the `CamelRouteVisualEntity` with `KameletVisualEntity`